### PR TITLE
The 'A Book A Month Page' is now opening properly

### PR DESCRIPTION
**Pull request by Priyanka-Kaintura**
Targets #2985

### Description
The 'A Book A Month' page is now opening properly.

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php [hunk 1]
```diff
@@ -318,6 +318,7 @@ public function unselectBookFromCurrentMonth(Book $book)
 
     public function bookAMonthIndex()
     {
+        $books = Book::all();
```
Comment (Ayush8923) [inline app/Http/Controllers/KnowledgeCafe/Library/BookController.php:321]: We do not need to show all the books on this page. On this page, we need to show the books corresponding to the user.

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php [hunk 2]
```diff
@@ -318,6 +318,7 @@ public function unselectBookFromCurrentMonth(Book $book)
...
         $booksCollection = BookAMonth::all()
             ->groupBy(function ($item) {
                 return Carbon::parse($item->created_at)->format('Y');
```
Comment (Ayush8923) [inline app/Http/Controllers/KnowledgeCafe/Library/BookController.php:324]: Please check how this query is working. And debug whether we are getting data shown here or not. and accordingly, changes need to be made in the frontend.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some comments.

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php (hunk 1 above)
Comment (Priyanka-Kaintura) [reply app/Http/Controllers/KnowledgeCafe/Library/BookController.php:321]: But sir, the books corresponding to the user comes from the '$bookCollection' variable, whereas this 'books' variable is to count the wishlisted and borrowed books.

Comment (Ayush8923) [reply app/Http/Controllers/KnowledgeCafe/Library/BookController.php:321]: Yes yes, you are right. 
But, when I test this PR, getting this error, then get the below error. Please check and test it locally.

image

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Left some comment.

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php (hunk 1 above)
Comment (Ayush8923) [reply app/Http/Controllers/KnowledgeCafe/Library/BookController.php:321]: Yes yes, you are right. 
But, when I test this PR, getting this error, then get the below error. Please check and test it locally.

image

Comment (Priyanka-Kaintura) [reply app/Http/Controllers/KnowledgeCafe/Library/BookController.php:321]: Sir it seem to work fine on my local system. Can you share a screeshot of line 29-46 of 'resource/views/KnowledgeCafe/library/books/book-a-month.blade.php' from where you are testing.

image

Comment (Ayush8923) [reply app/Http/Controllers/KnowledgeCafe/Library/BookController.php:321]: Okay, Got it. 
I tested this PR in a different system, it's working fine.

**Review by Ayush8923 — APPROVED** [review]

**Pull request merged**